src/components/UserList.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { useState, useRef, useEffect } from 'react';
+import { User } from '../types/User';
+import styles from './UserList.module.scss';
+import EllipseIcon from '../assets/ellipse.svg';
+import ListIcon from '../assets/list.svg';
+import GridIcon from '../assets/grid.svg';
+import { useNavigate, useLocation } from 'react-router-dom';
+import UserCard from './UserCard';
+
+type ViewMode = 'grid' | 'table';
+
+type UserListProps = {
+  users: User[];
+  loading: boolean;
+  error: string | null;
+  searchTerm: string;
+  setSearchTerm: (term: string) => void;
+};
+
+const UserList = ({ users, loading, error, searchTerm, setSearchTerm }: UserListProps) => {
+  const navigate = useNavigate();
+  const location = useLocation();
+  const [viewMode, setViewMode] = useState<ViewMode>(location.state?.previousViewMode || 'grid');
+  const userRefs = useRef<{ [key: string]: HTMLDivElement | null }>({});
+
+  useEffect(() => {
+    const scrollToId = location.state?.scrollToId;
+    if (scrollToId && userRefs.current[scrollToId]) {
+      userRefs.current[scrollToId]?.scrollIntoView({ behavior: 'smooth' });
```

Comment:
this scrolls (seemingly) a bit more than it should - a part of the user card is beneath the header - in case of table view the user row is not visible at all